### PR TITLE
[CN-675]: JVM GC Configuration

### PR DESCRIPTION
**Pull request by kutluhanmetin**
## Description

Enables to set JVM GC Configuration using Hazelcast Platform Operator.

## User Impact

- Users can enable GC logging.
- Users can set GC (Serial, Parallel, G1)
- Users can pass arbitrary JVM GC Arguments.

### CODE REVIEW

File: api/v1alpha1/hazelcast_types.go [hunk 1]
```diff
@@ -611,6 +622,59 @@ func (c *JVMMemoryConfiguration) GetMaxRAMPercentage() string {
...
 
+// JVMGCConfiguration is for configuring JVM Garbage Collector
+type JVMGCConfiguration struct {
+	// Logging enables logging when set to true
+	// +optional
+	Logging *bool `json:"logging,omitempty"`
+
+	// Collector is the Garbage Collector type
+	// +optional
+	Collector *string `json:"collector,omitempty"`
```
Comment (mtyazici) [inline api/v1alpha1/hazelcast_types.go:633]: I would suggest we put GCType here and not a string

**Review by mtyazici — APPROVED** [review]
LGTM, are we planning to add phone home or integration tests for GC config?

File: api/v1alpha1/hazelcast_types.go (hunk 1 above)
Comment (cagric0) [reply api/v1alpha1/hazelcast_types.go:633]: I agree with Muhammet. We should change its type to enum.

**Comment by cagric0** [comment]
> LGTM, are we planning to add phone home or integration tests for GC config?

We should discuss it with the team but I think integration tests are a must-have. I don't have a strong opinion about phone-home.

**Review by cagric0 — APPROVED** [review]

**Pull request merged**